pkg/controller/master/image/vm_image_controller.go

```diff
@@ -105,63 +113,77 @@ func (h *vmImageHandler) initialize(image *harvesterv1.VirtualMachineImage) (*ha
 		return image, err
 	}
 
-	image, err := h.checkImage(image)
+	toUpdate := image.DeepCopy()
+	toUpdate, err := h.checkImage(toUpdate)
 	if err != nil {
-		logrus.Errorf("failed to check image %s/%s, err: %v", image.Namespace, image.Name, err)
-		h.imageController.EnqueueAfter(image.Namespace, image.Name, 10*time.Second)
-		return h.images.Update(image)
+		logrus.WithError(err).WithFields(logrus.Fields{
+			"namespace": toUpdate.Namespace,
+			"name":      toUpdate.Name,
+		}).Error("failed to check vmimage")
+		h.imageController.EnqueueAfter(toUpdate.Namespace, toUpdate.Name, 10*time.Second)
```

Comment:
Both those 2 will trigger 'OnChange' again (suppose the `Update` really updates the object), which one is preferred?

Reply:
Thanks for the comment ! Like you said, currently the onChange trigger twice in the above code snippet. One appropriate way to deal with this is check if Update works successfully, if not, we should invoke EnqueueAfter.  I'll address this later. :smiley: